### PR TITLE
Add Custom Error Handling page (vLatest)

### PR DESCRIPTION
**Pull request by tarekoraby**
cp #978

### CODE REVIEW

**Comment by jouni** [comment]
Why is there two articles with the same content but with different titles?

https://pr-994.dcmp4ajsd49kq.amplifyapp.com/flow/advanced/custom-error-handler

https://pr-994.dcmp4ajsd49kq.amplifyapp.com/flow/advanced/session-and-ui-init-listener

**Comment by tarekoraby** [comment]
Because I’m a dum-dum and cherry picked the wrong content.

**Comment by tarekoraby** [comment]
Ah, it turned out I committed blindly before hitting save. Now the right content should be there.

File: articles/flow/advanced/custom-error-handler.asciidoc [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+title: Custom Error Handling
+order: 145
+layout: page
+---
+
+= Custom Error Handling
+
+You can customize the error views related to navigation between different routes as described in the <<{articles}/flow/routing/exceptions#, Router Exception Handling>> section.
+
+For other unhandled runtime exceptions, an `ErrorHandler` class can be used to gracefully let users know that something went wrong. In order to do so, first create your custom `ErrorHandler` class, then use that class to override the default error handler via `VaadinSession.getCurrent().setErrorHandler(ErrorHandler)`.
```
Comment (jouni) [inline articles/flow/advanced/custom-error-handler.asciidoc:11]:
```suggestion
For other unhandled runtime exceptions, an [classname]#ErrorHandler# class can be used to gracefully let users know that something went wrong.
In order to do so, first create your custom [classname]#ErrorHandler# class, then use that class to override the default error handler.
```

File: articles/flow/advanced/custom-error-handler.asciidoc [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+For example, the following `CustomErrorHandler` class logs the error and displays a friendly notification to the user.
```
Comment (jouni) [inline articles/flow/advanced/custom-error-handler.asciidoc:13]:
```suggestion
For example, the following [classname]#CustomErrorHandler# class logs the error and displays a friendly notification to the user.
```

File: articles/flow/advanced/custom-error-handler.asciidoc [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+            UI.getCurrent().access(() -> {
+                Notification.show("An internal error has occurred." +
+                        "Please contact support.");
+            });
+        }
+    }
+}
+----
+
+In order for the logic in this class to take effect, you can then assign it to the current user session using `VaadinSession.getCurrent().setErrorHandler(new CustomErrorHandler());`. In order to apply the custom error-handling logic to all user sessions, it is usually more convenient to make use of a `SessionInitListener`, which receives an event each time a new `VaadinSession` is initialized (see <<session-and-ui-init-listener#,Session and UI Listeners tutorial>> to learn how to create a `SessionInitListener`).
```
Comment (jouni) [inline articles/flow/advanced/custom-error-handler.asciidoc:34]: Inlining code snippets within paragraph text is not something I encourage to do. Instead, when you write them as 
explicit source code blocks, users get syntax highlighting and the copy-to-clipboard button. And it’s likely easier to skim the page and notice the relevant code you need, when it’s not "hiding" inside a paragraph.

```suggestion
Then assign the custom error handler to the current user session:

[source,java]
----
VaadinSession.getCurrent().setErrorHandler(new CustomErrorHandler());
----
 
To apply the custom error handler to all user sessions, you can use a [classname]#SessionInitListener#, which receives an event each time a new `VaadinSession` is initialized.
See the <<session-and-ui-init-listener#,Session and UI Listeners>> article to learn how to create a [classname]#SessionInitListener#.
```

**Review by jouni — COMMENTED** [review]
Small language and formatting improvements.

**Review by jouni — APPROVED** [review]

**Pull request merged**